frontend/gateway/grpcclient/client.go

```diff
@@ -310,18 +311,30 @@ func (c *grpcClient) Solve(ctx context.Context, creq client.SolveRequest) (*clie
 	} else {
 		res.Metadata = resp.Result.Metadata
 		switch pbRes := resp.Result.Result.(type) {
-		case *pb.Result_Ref:
-			if id := pbRes.Ref; id != "" {
+		case *pb.Result_RefDeprecated:
+			if id := pbRes.RefDeprecated; id != "" {
 				res.SetRef(&reference{id: id, c: c})
 			}
-		case *pb.Result_Refs:
-			for k, v := range pbRes.Refs.Refs {
+		case *pb.Result_RefsDeprecated:
+			for k, v := range pbRes.RefsDeprecated.Refs {
 				ref := &reference{id: v, c: c}
 				if v == "" {
 					ref = nil
 				}
 				res.AddRef(k, ref)
 			}
+		case *pb.Result_Ref:
+			if ids := pbRes.Ref.Ids; len(ids) > 0 {
+				res.SetRef(&reference{id: ids[0], c: c})
```

Comment:
Originally, I wanted to just change the types to `[]string` but only handle single element, but I think we should change these to support multi-result arrays right? i.e.
```go
res.SetRef(&reference{id: ids, c: c})
```